RAGchain/DB/mongo_db.py

```diff
@@ -54,19 +55,31 @@ def create_or_load(self):
         else:
             self.create()
 
-    def save(self, passages: List[Passage]):
+    def save(self, passages: List[Passage], upsert: bool = False):
```

Comment:
add upsert option to base abstractmethod.
And make upsert option for pickle DB as well.